drivers/power/ltc3337/ltc3337.c

```diff
@@ -0,0 +1,715 @@
+{
+	uint16_t reg_value;
+	int ret;
+
+	//Input parameter NULL check
+	if(!dev)
+		return -EINVAL;
+
+	reg_value = dev->latched_reg_a;
+
+	if(shutdown_en)
+		reg_value |= LTC3337_RA_SHTDN;
+	else
+		reg_value &= ~(LTC3337_RA_SHTDN);
+
+	ret = ltc3337_write_reg(dev, LTC3337_REG_A, reg_value);
+	if(ret)
+		return ret;
+
+	dev->latched_reg_a = reg_value;
+
+	return 0;
+}
+
+/**
+ * Sets the alarm level for the coulomb counter alarm. The input value is the
+ * raw register value, and can be calculated using LTC3337_CalculateChargeRegister
+ * or manually by hand.
+ * IMPORTANT: Only the upper 8-bits of the value are utilized. Use the roundUp
+ * flag to have the driver optionally round up to the next valid bit
+ * @param dev - Device to configure
+ * @param counter_reg_val - Register value, as if all 16-bits were utilized
+ * @param round_up - 1 -Have the driver round up to next usable alarm bit
+ * @returns 0 on success, Non-0 on error
+ */
+int ltc3337_set_counter_alarm(struct ltc3337_dev* dev, uint16_t counter_reg_val,
+			      uint8_t round_up)
+{
+	uint8_t data_to_set;
+	uint16_t reg_value;
+	int ret;
+
+	//Input parameter NULL check
+	if(!dev)
+		return -EINVAL;
+
+	reg_value = dev->latched_reg_a;
+
+	//Only the upper 8 bits are utilized
+	data_to_set = (counter_reg_val >> 8);
+
+	//If rounding was requested, check the if any of the lower bits are set,
+	//and if we wont overflow dataToSet beyond 8-bits, increment
+	if((round_up) && (counter_reg_val & 0xFF) && (data_to_set != 0xFF))
+		data_to_set++;
+
+	reg_value &= ~(LTC3337_RA_ALARM_LVL_MSK);
+	reg_value |= no_os_field_prep(LTC3337_RA_ALARM_LVL_MSK, data_to_set);
+
+	ret = ltc3337_write_reg(dev, LTC3337_REG_A, reg_value);
+	if(ret)
+		return ret;
+
+	dev->latched_reg_a = reg_value;
+
+	return 0;
+}
+
+/**
+ * Sets the level for the coulomb counter accumulated charge register.
+ * The input value is the raw register value, and can be calculated using
+ * LTC3337_CalculateChargeRegister or manually by hand.
+ * IMPORTANT: Only the upper 8-bits of the value are utilized. Use the roundUp
+ * flag to have the driver optionally round up to the next valid bit
+ * @param dev - Device to configure
+ * @param reg_value - Register value, as if all 16-bits were utilized
+ * @param round_up - 1 -Have the driver round up to next usable alarm bit
+ * @returns 0 on success, Non-0 on error
+ */
+int ltc3337_set_accumulated_charge(struct ltc3337_dev* dev, uint16_t reg_value,
+				   uint8_t round_up)
+{
+	uint16_t data_to_set;
+
+	//Input parameter NULL check
+	if(!dev)
+		return -EINVAL;
+
+	//Only upper 8-bits (MSB) are writable
+	data_to_set = reg_value & 0xFF00;
+
+	//If rounding up was requested, check if any of the lower 8-bits of the
+	//original value are set, and verify dataToSet will not overflow 16-bits
+	//if we increment
+	//Add 1 to the MSB of the data to set to account for rounding
+	//request
+	if((round_up) && (reg_value & 0xFF) && (data_to_set != 0xFF00))
+		data_to_set += 0x100;
```

Comment:
Add macro for 0x100 i.e. LTC3337_DATA_INCREMENT

Reply:
This was mentioned in a previous comment from danmois above.  This block of code which uses 0x100 and 0xFF and 0xFF00 has no significant numerical value related to the device, except we are isolating whether there is data in the MSB or LSB of a uint16_t.  In my opinion making these macros would make the code harder to follow, where as the 0xFF 0xFF00 and 0x100 are intuitive values, with provided comments describing the logic.

Will make the changes if necessary...